plugin/ui/webpack.config.js

```diff
@@ -0,0 +1,121 @@
+const path = require('path');
+const CopyPlugin = require('copy-webpack-plugin');
+const ModuleFederationPlugin = require('webpack/lib/container/ModuleFederationPlugin');
+const pjson = require('./package.json');
+const ESLintWebpackPlugin = require('eslint-webpack-plugin');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
+const pluginName = pjson.name;
+
+const config = {
+  entry: path.resolve(__dirname, './src'),
+  output: {
+    path: path.resolve(__dirname, 'build'),
+    filename: '[name].app.[contenthash:8].js',
+    publicPath: 'auto',
+    clean: true,
+  },
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/,
+      },
+      {
+        test: /\.(sa|sc)ss$/,
+        exclude: /node_modules/,
+        use: [
+          'style-loader',
+          {
+            loader: 'css-loader',
+            options: {
+              modules: {
+                localIdentName: `${pluginName}_[name]__[local]--[hash:base64:5]`,
+              },
+            },
+          },
+          'sass-loader',
+          {
+            loader: 'sass-resources-loader',
+            options: {
+              resources: path.resolve(__dirname, './src/common/css/**/*.scss'),
+            },
+          },
+        ],
+      },
+      {
+        test: /\.svg$/,
+        loader: 'svg-inline-loader',
+      },
+    ],
+  },
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js', '.sass', '.scss', '.css'],
+    alias: {
+      components: path.resolve(__dirname, 'src/components'),
+      constants: path.resolve(__dirname, 'src/constants'),
+      icons: path.resolve(__dirname, 'src/icons'),
+      hooks: path.resolve(__dirname, 'src/hooks'),
+      utils: path.resolve(__dirname, 'src/utils'),
+      analyticsEvents: path.resolve(__dirname, 'src/analyticsEvents'),
+    },
+  },
+  externals: ['redux'],
+  plugins: [
+    new ForkTsCheckerWebpackPlugin({ typescript: { memoryLimit: 4096 } }),
+    new ESLintWebpackPlugin({
+      context: './src',
+      extensions: ['.scss', '.css', '.ts', '.tsx'],
+      threads: true,
+    }),
+    new ModuleFederationPlugin({
+      name: 'plugin_name',
+      filename: `remoteEntity.js`,
+      shared: {
+        react: {
+          import: 'react',
+          shareKey: 'react',
+          shareScope: 'default',
+          singleton: true,
+          requiredVersion: pjson.dependencies['react'],
```

Comment:
As I see it is not the best solution here, refer this [issue](https://github.com/webpack/webpack/issues/15971#issuecomment-1172163299).
But I guess you've fixed the warnings in the browser console related to the deps sharing. Am I right?

Reply:
Yes, you are right!

Reply:
Ok, let's bypass this warnings.